### ISSUE TITLE
 本地运行 invoke 可以跳过缓存验证吗？

### ISSUE THREAD

**Issue by bear-ei** [midwayjs/midway-faas#126]
在项目中使用typeorm。     

invoke本地运行会导致TypeORM的单例实例陷入混乱，无法重用链接。

应该和下面这个问题是类似的。

https://github.com/typeorm/typeorm/issues/3427#issuecomment-549770047

A long time has passed, and I finally had some time to actually look into this. It appears that this isn't really a problem in the AWS runtime at all, merely when using sls-offline to locally debug.

The reason (and fix) for that are mentioned in #2623 (which this issue is a duplicate of).

tl;dr: If you're using sls-offline, please invoke it with and none of the patches mentioned in this issue will be required. It's always been fine in the actual runtime (AWS) itself.--skipCacheInvalidation

What basically happens is that the singleton instance of TypeORM gets messed with when using the shared code runner from sls-offline.

@pleerock If more people confirm this is effective, I think you can close this issue.

**Comment by echosoar** [comment]
+ f invoke 命令进行本地调用是单次的，在执行结束后会关闭当前进程，在多次调用之间是没有缓存的。
+ 单独使用 serverless-invoke 包进行单测中的函数调用，对于invoke自身来说，是没有用户代码缓存的，但是不排除引用的包里面存在自身的缓存逻辑，这一部分在invoke层面不好解决。

**Comment by bear-ei** [comment]
抱歉，忘了说。

不是使用f invoke命令，是执行单元测试。

多个测试一起执行的时候输出结果可以看到没有关闭当前进程，也没有触发onStop的生命周期。

**Comment by czy88840616** [comment]
缓存能给个示例么？ onStop 的确，因为测试的 light-runtime 没有走 close 方法，所以不会触发。

**Comment by bear-ei** [comment]
demo   https://github.com/SoundShell/midway-faas-typeorm

```
$ NODE_ENV=local f test
Testing all *.test.js/ts...


  /test/index.test.ts
ConnectionManager:false     // 第一次执行，typeorm链接不存在，建立链接正常完成单元测试
    ✓ invoke-1 (3997ms)
ConnectionManager:true     // 第二次执行，检查到typeorm链接存在
ConnectionManager:true
    1) invoke-2


  1 passing (7s)
  1 failing

  1) /test/index.test.ts
       invoke-2:
     EntityMetadataNotFound: No metadata for "IndexEntity" was found.   // 复用链接失败，找不到实体
```

推测和 typeorm/typeorm#3427 问题一致，是因为本地运行缓存导致。

**Comment by bear-ei** [comment]
> 在项目中使用typeorm。
> 
> invoke本地运行会导致TypeORM的单例实例陷入混乱，无法重用链接。
> 
> 应该和下面这个问题是类似的。
> 
> [typeorm/typeorm#3427 (comment)](https://github.com/typeorm/typeorm/issues/3427#issuecomment-549770047)
> 
> A long time has passed, and I finally had some time to actually look into this. It appears that this isn't really a problem in the AWS runtime at all, merely when using sls-offline to locally debug.
> 
> The reason (and fix) for that are mentioned in #2623 (which this issue is a duplicate of).
> 
> tl;dr: If you're using sls-offline, please invoke it with and none of the patches mentioned in this issue will be required. It's always been fine in the actual runtime (AWS) itself.--skipCacheInvalidation
> 
> What basically happens is that the singleton instance of TypeORM gets messed with when using the shared code runner from sls-offline.
> 
> @pleerock If more people confirm this is effective, I think you can close this issue.
